src/Compilers/Core/Portable/MetadataReader/PEModule.cs

```diff
@@ -1921,6 +1930,26 @@ internal static bool CrackStringArrayInAttributeValue(out ImmutableArray<string?
             return false;
         }
 
+        private static bool ValidateMetadataArrayLength(uint length)
+        {
+            // Null arrays are represented in metadata by a length of 0xFFFF_FFFF. See ECMA 335 II.23.3.
+            const uint NullArray = 0xFFFF_FFFF;
+
+            if (length == NullArray)
+            {
+                return false;
+            }
+
+            // II.23.3 says that the length must be a int32 if not null. There can't be negative lengths,
```

Comment:
>// II.23.3 says that the length must be a int32 if not null [](http://example.com/codeflow?start=12&length=59)

I cannot find where does it say that. Here is what I found: "NumElem is an **unsigned int32** specifying the number of elements"